### PR TITLE
Don't use JvmField and JvmStatic in non-interop mode

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: wire-grpc-tests/src/test/proto-grpc/routeguide/Feature.kt [hunk 1]
```diff
@@ -51,7 +50,6 @@ data class Feature(
   }
 
   companion object {
-    @JvmField
     val ADAPTER: ProtoAdapter<Feature> = object : ProtoAdapter<Feature>(
```
Comment (JakeWharton) [inline wire-grpc-tests/src/test/proto-grpc/routeguide/Feature.kt]: This is still looked up via reflection in ProtoAdapterJvm

File: wire-moshi-adapter/src/test/java/com/squareup/wire/protos/person/kotlin/Person.kt [hunk 2]
```diff
@@ -140,14 +137,12 @@ data class Person(
...
 
     companion object {
-      @JvmField
       val ADAPTER: ProtoAdapter<PhoneType> = object : EnumAdapter<PhoneType>(
         PhoneType::class
       ) {
         override fun fromValue(value: Int): PhoneType = PhoneType.fromValue(value)
       }
 
-      @JvmStatic
```
Comment (JakeWharton) [inline wire-moshi-adapter/src/test/java/com/squareup/wire/protos/person/kotlin/Person.kt]: This is required for RuntimeEnumAdapter

**Comment by Egorand** [comment]
I'll fix reflection to look for both static fields and companions to support both modes.

**Comment by JakeWharton** [comment]
Is there a motivation to remove these? They don't prevent the generated code from being multiplatform.

**Comment by Egorand** [comment]
IDE complained about them when I tried compiling against JS

**Comment by JakeWharton** [comment]
Are you using kotlin2js or the multiplatform plugin? We have these in Okio's common code, for example, which compiles to JS just fine.

**Comment by Egorand** [comment]
As discussed offline, will punt this for now, JVM annotations should work fine if generated code is used by a multiplatform module. In future we might enable the compiler to produce JS or Native-specific code.

**Pull request closed without merging**